app/Http/Controllers/StorageProviderController.php

```
@@ -60,6 +65,32 @@ public function store(Request $request): RedirectResponse
+
+    #[Get('/dropbox/callback', name: 'storage-providers.dropbox.callback')]
+    public function dropboxCallback(Request $request, ConnectDropbox $action): RedirectResponse
+    {
+        $this->authorize('create', StorageProvider::class);
+
+        try {
+            $action->handleCallback(user(), $request);
+        } catch (ValidationException $e) {
+            return to_route('storage-providers')->with('error', $e->validator->errors()->first());
+        } catch (Throwable $e) {
+            Log::error('Dropbox OAuth callback failed', ['error' => $e->getMessage()]);
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'dropboxCallback'. (undefined)

(StaticAccess)